.github/workflows/python-e2e-test.yml

```diff
@@ -29,23 +33,37 @@ jobs:
         with:
           python-version: '${{ matrix.python-version }}'
 
-      - name: Install poetry
-        uses: abatilo/actions-poetry@7b6d33e44b4f08d7021a1dee3c044e9c253d6439 # pin@v3
-        with:
-          poetry-version: "1.7.1"
+      - name: Install uv
+        run: curl -LsSf https://astral.sh/uv/0.4.7/install.sh | sh
+
+      # It's OK to use the default python version and not the specified one
+      - name: Install all projects dependencies
+        working-directory: python
+        run: uv sync --all-extras --dev
 
-      - name: Install the project dependencies
+      - name: Build the python package
         working-directory: python
-        run: poetry build
+        run: uv run ./scripts/build_python_package.py
 
       - name: List wheels
         working-directory: python
         run: python -c "import os; print('\n'.join(os.listdir('dist')))"
 
-      - name: Install magika
+      - name: Install the magika package
         working-directory: python
         run: python -m pip install $(python -c "import glob; print(glob.glob('dist/magika-*.whl')[0])")
         shell: bash # Allows for cross-platform
 
+      - name: Run magika --version
+        working-directory: python
+        run: magika --version
+        shell: bash # Allows for cross-platform
+
       - name: Run magika with tests_data
-        run: magika -r tests_data/ || exit 1
+        run: magika -r tests_data/basic || exit 1
```

Comment:
nit: `|| exit 1` should be redundant, GitHub should run with `set -e`.

https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell

Reply:
fixed